modules/ztp-image-based-upgrade.adoc

```diff
@@ -0,0 +1,71 @@
+[id="ztp-image-based-upgrade-concept_{context}"]
+= Image-based upgrade for {sno} cluster with the {lcao}
+
+You can upgrade the platform version of a managed {sno} cluster with the {lcao}.
+Using a generated seed image allows you to skip multiple minor releases and it is faster than the standard upgrade method.
+
+You generate the seed image from a {sno} seed cluster, which is deployed and configured with the desired {product-title} and applications. You can upgrade the platform version on any {sno} that has the same combination of hardware, Day 2 Operators, and cluster configuration.
+
+[IMPORTANT]
+====
+The image-based upgrade uses custom images that are specific to the hardware platform that the clusters are running on.
+Each different hardware platform requires a separate custom image.
+====
+// TODO agree on how much we should specify this. Is this enough or detail about CPU topology, deployment method, etc.
+
+The {lcao} uses two custom resources (CRs) on the participating clusters:
+
+* On the seed cluster, the `SeedGenerator` CR allows the seed image generation. This CR specifies the repository to push the seed image to.
+* On the target cluster, the `ImageBasedUpgrade` CR specifies the seed container image for the upgrade of the target cluster and the backup configurations for your workloads.
+
+After generating the seed image on the seed cluster, you can move throught the stages on the target cluster by setting the `spec.stage` field to the following values in the `ImageBasedUpgrade` CR:
+
+* `Idle`
+* `Prep`
+* `Upgrade`
+* `Rollback` (Optional)
+* `Idle`
+
+[discrete]
+== Idle stage
+
+This is the default stage when the {lcao} is installed. There is no ongoing upgrade and the cluster is ready to move to the `Prep` stage.
+
+After a successful upgrade or a rollback, you commit to the change by manually moving the stage to `Idle`.
+Changing to this stage ensures that the {lcao} cleans up resources, so the cluster is ready for upgrades.
+
+[discrete]
+== Prep stage
+
+You can complete this stage before a scheduled maintenance window.
+
+In the `Prep` stage, the {lcao} ensures that the target clusters are ready to proceed to the `Upgrade` stage by checking if they meet certain conditions.
+The Operator uses the `ostree` versioning system and the OADP Operator to create a backup, which allow complete cluster reconfiguration after upgrade or rollback.
+
+At the `Prep` stage, the {lcao} pulls the seed image and additional container images specified in the seed image.
+
+You can stop the upgrade process. If you stop, the Operator performs cleanup operations and sets the upgrade stage back to `Idle`.
+
+[discrete]
+== Upgrade stage
+
+Just before the {lcao} starts the upgrade process, the OADP Operator creates a backup of the user-specified resources on a compatible Object storage solution, such as an Amazon S3 bucket.
+These OADP `Backup` and `Restore` CRs are applied to the target cluster after the upgrade.
```

Comment:
Well, the backup is done during the upgrade stage before booting in the new stateroot. But the recover CR is executed once the new version of the SNO is ready, and once the platform is ready, then recover the apps.

Reply:
I rearranged and rephrased this part, also emphasizing that backup is optional. WDYT?
> Before the {lcao} starts the upgrade process, you can create an optional backup of your cluster resources with the OADP Operator.
The OADP Operator creates a backup of the user-specified resources on a compatible Object storage solution, which are defined in the `Backup` CR.
After the target cluster reboots with the new platform version, the Operator applies the cluster and application configurations defined in the `Restore` CR.